cloud/microsoft/office365/exchange/mode/emailactivity.pm

```diff
@@ -34,11 +35,14 @@ sub custom_active_perfdata {
         $total_options{cast_int} = 1;
     }
 
-    $self->{output}->perfdata_add(label => 'active_mailboxes',
+    $self->{result_values}->{report_date} =~ /^([0-9]{4})-([0-9]{2})-([0-9]{2})$/;
+    $self->{output}->perfdata_add(label => 'perfdate', value => timelocal(0,0,12,$3,$2-1,$1-1900));
```

Comment:
Can you use a more clear metric name like 'report_date', and maybe add a new label also.

Reply:
Office365 API reports daily data, so at least from the day before, sometimes from several days ago, depending on how fast Microsoft generates reports at this moment...
This `perfdate` metric is then set to the _Report Refresh Date_ from the API result.
The only purpose of this metric is to be handled by Stream Connector Scripts to properly set the data timestamp (see https://github.com/centreon/centreon-stream-connector-scripts/pull/23). So that data is sent to InfluxDB (for instance) with the proper report date, not the request date... Leading to accurate graphs as shown [above](https://github.com/centreon/centreon-plugins/pull/2042#issuecomment-665525294).
`perfdate` then sounds ideal for this purpose, as the _perfdata_ date.
As a "meta" metric, it does not need its own graph as any other proper metric does, thus does not need a new label.
Btw, note that InfluxDB Stream Connector Script rejects metrics which are not new-label named, which is nice, `perfdate` will not unnecessarily reach InfluxDB.
Finally, this `perfdate` metric could be used in any other mode facing the same non-realtime data behaviour.